src/useForm.ts

```diff
@@ -90,10 +93,25 @@ export function useForm(
     };
 
     // Append extra data from config
+    if (typeof extraData === 'function') {
+      extraData = (extraData as (() => DataObject | Promise<DataObject>)).call(
+        null
+      );
+      if (extraData instanceof Promise) {
+        extraData = await extraData;
+      }
+    }
+
```

Comment:
Don't you need to iterate through the individual entries in `extraData`? There could be one key-value that's a promise, but another that's a data object.

Reply:
That's handled further down.